app/lib/service/download_counts/computations.dart

```diff
@@ -6,15 +6,45 @@ import 'dart:math';
 
 import 'package:_pub_shared/data/download_counts_data.dart';
 import 'package:gcloud/storage.dart';
+import 'package:pub_dev/package/backend.dart';
 import 'package:pub_dev/service/download_counts/backend.dart';
 import 'package:pub_dev/service/download_counts/download_counts.dart';
 import 'package:pub_dev/service/download_counts/models.dart';
+import 'package:pub_dev/service/download_counts/package_trends.dart';
 import 'package:pub_dev/shared/configuration.dart';
 import 'package:pub_dev/shared/storage.dart';
 import 'package:pub_dev/shared/utils.dart';
 
 import '../../shared/redis_cache.dart' show cache;
 
+Future<void> computeTrendScoreTask() async {
+  final trendScores = await computeTrend();
+  await uploadTrendScores(trendScores);
+}
+
+Future<Map<String, double>> computeTrend() async {
+  final res = <String, double>{};
+
+  await for (final pkg in packageBackend.allPackages()) {
+    final name = pkg.name!;
+    final downloads =
+        (await downloadCountsBackend.lookupDownloadCountData(name))
```

Comment:
You could consider a retry around this part, so that `computeTrend` doesn't fail just because a single lookup here failed.

Indeed, you could also consider ignore errors here.. or maybe ignore up-to 50 errors, so if there is less than 50 errors you just ignore them (log them), but don't fail the computation.

Just an idea, not sure it's smart.

Reply:
or mabye `lookupDownloadCountData` already does retries